File: python/questions/maxArea.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        ans = 0
        h = 0
        w = 0
        while l < r:
            w = r - l
            if height[r] > height[l]:
                h = height[l]
                l += 1
            else:
                h = height[r]
                r -= 1
            ans = max(ans, h*w)
        return ans

File: cpp/questions/maxArea.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0;
        int r = height.size() - 1;
        int ans = 0;
        while (l < r) {
            bool rbl = height[r] > height[l];
            int minh = rbl ? height[l] : height[r];
            ans = std::max(ans, (r-l)*minh);
            r -= rbl ? 0 : 1;
            l += rbl ? 1 : 0;
        }
        return ans;
    }
};
